db/memtable.cc

```diff
@@ -574,6 +585,8 @@ bool MemTable::Add(SequenceNumber s, ValueType type,
   }
   if (type == kTypeRangeDeletion) {
     is_range_del_table_empty_.store(false, std::memory_order_relaxed);
+
+    RebuildFragmentedTombstones(ReadOptions());
```

Comment:
Rather than rebuild the fragmented tombstones on every update, you could invalidate the cached fragmented tombstones here and rebuild them the next time a new range tombstone iterator is created. It might be sufficient to simply set `fragmented_tombstone_list = nullptr` here, but I haven't thought through whether there are any gotchas with that.

Hmm, one gotcha with this approach is how you construct the fragmented tombstones in `NewRangeTombstoneIterator`.